packages/gis/lua/Layer.lua

```diff
@@ -1062,6 +1114,168 @@ Layer_ = {
+		end)
+
+		forEachElement(dataAttributes, function(_, attr)
+			forEachElement(temporalAttributes, function(time)
+			table.insert(temporalAttributes[time], {prefix = attr, sufix = "", name = attr})
+			end)
+		end)
+
+		forEachElement(temporalAttributes, function(time, attributeList)
+			local attrNames = {}
+			local mapAttributes = {}
+			forEachElement(attributeList, function(_, attr)
+				table.insert(attrNames, attr.prefix)
+				mapAttributes[attr.name] = attr.prefix
+			end)
+
+			local dset = TerraLib().getDataSet(self.project, self.name)
+			local toSet = {}
+			for i = 0, #dset do
+				toSet[i+1] = {}
+				for k, v in pairs(dset[i]) do
+					toSet[i+1][k] = nil
+					if mapAttributes[k] then
+						toSet[i+1][mapAttributes[k]] = v
+					end
+				end
+			end
+
+			local newLayerName = self.name.."_"..time
+			local fileName = newLayerName.."."..self.source
+			-- a temp layer is needed because currently one can't remove attributes from a layer thus this layer holds only the desired attributes
+			local tempLayer = "_l_y_r_"
+			TerraLib().saveDataSet(self.project, self.name, toSet, tempLayer, attrNames)
+			local from = {project = self.project, layer = tempLayer}
+			local to = {file = fileName, type = self.source}
+			TerraLib().saveLayerAs(from, to, true, attrNames)
+			Layer{project = self.project, name = newLayerName, file = fileName}
+			TerraLib().removeLayer(self.project, tempLayer)
+		end)
+	end,
+	--- Merges every temporal layer from the layer's project into a single layer with all temporal attribute.
+	-- Atributes from temporal layers are created with the attribute name followed by the time from its layer.
+	-- Every temporal layer must have the same geometry.
+	-- @usage -- DONTRUN
+	-- layer:merge()
+	merge = function(self)
+		local newLayerName, time = string.match(self.name, "(.+)_(%d+)")
+		if not newLayerName or not time then
+			customError("Layer '"..self.name.."' is not a temporal layer.")
+		end
+
+		if self.project.layers[newLayerName] then
+			customWarning("Layer '"..newLayerName.."' already exisists.")
+		end
+
+		local temporalLayers = {}
+		local attributes = {}
+		forEachElement(self.project.layers, function(layerName)
+			local prefix, sufix = string.match(layerName, "(.+)_(%d+)")
+			if prefix and sufix and prefix == newLayerName then
+				temporalLayers[layerName] = true
+				local attribs = TerraLib().getPropertyNames(self.project, layerName)
+				forEachElement(attribs, function(_, attribute)
+					if not attributes[attribute] then
+						attributes[attribute] = {}
+					end
+
+					table.insert(attributes[attribute], {layer = layerName, time = sufix})
+				end)
+			end
+		end)
+
+		local dSetSize = #TerraLib().getDataSet(self.project, self.name)
+		forEachOrderedElement(temporalLayers, function(layer)
+			if dSetSize ~= #TerraLib().getDataSet(self.project, layer) then
+				customError("Layer '"..self.name.."' cannot merge with the layer '"..layer.."'.")
```

Comment:
Layer ... cannot be merged with [remove the] layer ... because they have different numbers of objects.